src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,357 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   Tweaked colour picker allows for tooltip texts translations and has a customized theming.
+/// </summary>
+public class TweakedColourPicker : ColorPicker
+{
+    /// <summary>
+    ///   This is where presets are stored after a colour picker exited scene tree.
+    /// <remarks>
+    ///   KeyValuePair&lt;GetPath(), GetPresets()&gt;:
+    ///   <ul>
+    ///     <li> First parameter (string) is the string format of NodePath; </li>
+    ///     <li> Second parameter (Color[]) is the preset colours to be stored. </li>
+    ///   </ul>
+    /// </remarks>
+    /// </summary>
+    private static readonly List<KeyValuePair<string, Color[]>> PresetsStorage
+        = new List<KeyValuePair<string, Color[]>>();
+
+    private readonly List<TweakedColourPickerPreset> presets = new List<TweakedColourPickerPreset>();
+
+    private HSlider sliderROrH;
+    private HSlider sliderGOrS;
+    private HSlider sliderBOrV;
+    private HSlider sliderA;
+    private ToolButton pickerButton;
+    private CheckButton hsvCheckButton;
+    private CheckButton rawCheckButton;
+    private LineEdit htmlColourEdit;
+    private HSeparator separator;
+    private GridContainer presetsContainer;
+    private TextureButton addPresetButton;
+
+    private bool hsvButtonEnabled = true;
+    private bool rawButtonEnabled = true;
+    private bool presetsEnabled = true;
+    private bool presetsVisible = true;
+
+    /// <summary>
+    ///   Decide if user can toggle HSV CheckButton to switch HSV mode.
+    /// </summary>
+    [Export]
+    public bool HSVButtonEnabled
+    {
+        get => hsvButtonEnabled;
+        set
+        {
+            hsvButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    /// <summary>
+    ///   Decide if user can toggle Raw CheckButton to switch Raw mode.
+    /// </summary>
+    [Export]
+    public bool RawButtonEnabled
+    {
+        get => rawButtonEnabled;
+        set
+        {
+            rawButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool HsvMode
```

Comment:
Shouldn't this be named `HSVMode` to be consistent with `HSVButtonEnabled`?

Reply:
This overrides Godot property HsvMode to avoid direct change to Godot's properties. So I have to name like this.

Reply:
Well maybe the other property that can be renamed, should be? To improve consistency.